.github/workflows/stale-bot.yml

```diff
@@ -0,0 +1,33 @@
+# Go to https://github.com/actions/stale for the full documentation of these options.
+
+name: 'Stale bot'
+on:
+  schedule:
+    - cron: '30 1 * * *'
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v3.0.18
+        with:
+          # Rate limit per run, defaults to 30.
+          operations-per-run: 30
```

Comment:
What does this mean for us? Will it only check 30 issues each time, or only comment or close 30 issues per run?

Reply:
From the stale bot docs:

> `operations-per-run`

> Used to limit the number of operations made with the GitHub API to avoid reaching the rate limit.
Based on your project, your GitHub business plan and the date of the cron job you set for this action, you can increase this limit to a higher number.

> When debugging, you can set it to a much higher number like 1000 since there will be fewer operations made with the GitHub API.
Only the actor and the batch of issues (100 per batch) will consume the operations.

> Default value: 30

Reference: https://github.com/actions/stale#operations-per-run

Reply:
The developer has a pending PR that clarifies this `operations-per-run` thing, copy-pasting without links back to their repo:

> Context:
> The action will perform some API calls to GitHub to fetch or close issues and pull requests, set or update labels, add comments, delete branches, etc.
> These operations are made in a very short period of time - because the action is very fast to run - and can be numerous based on your project action configuration and the quantity of issues and pull requests within it.
> GitHub has a rate limit and if reached will block all of these API calls for one hour (or API calls from other actions using the same user (a.k: the github-token from the repo-token option)).
> This option aim to avoid crossing this situation by limiting the number of operations for a single run.
> 
> Purpose:
> This option aim to limit the number of operations made with the GitHub API to avoid reaching the rate limit.
> 
> Based on your project, your GitHub business plan and the date of the cron job you set for this action, you can increase this limit to a higher number.
> If you are not sure which is the right value for you or if the default value is good enough, you could enable the logs and look at the end of the stale action.
> If you reached the limit, a warning will be visible and will suggest you to increase the number of operations.
> If you don't increase it, the risk is to have un-processed issues or pull requests, that's all.
> 
> When debugging, you can set it to a much higher number like 1000 since there will be fewer operations made with the GitHub API.
> Only the actor and the batch of issues (100 per batch) will consume the operations.
> 
> Default value: 30

Reply:
Seems that all stale bots have a limit like this though.

The Probot Stale app uses 30 operations per hour as the limit:

> ## Why did only some issues and pull requests get marked stale?
> To avoid triggering abuse prevention mechanisms on GitHub, only 30 issues and pull requests will be marked or closed per hour. If your repository has more than that, it will just take a few hours or days to mark them all.

https://github.com/probot/stale#why-did-only-some-issues-and-pull-requests-get-marked-stale

Reply:
Maybe let's run more often in case it's a limiting factor?

Are the labels used to limit the fetch or does it fetch all issues and then filter? If all issues we can expect ~8 queries if it fetches 100 per page and then 22 for comments or closing?

Reply:
> Maybe let's run more often in case it's a limiting factor?

I don't think this would be a good idea, as all of their examples use the current cron job cycle of once every day at `01:30` hours. If they wanted normal users to run the action more often, I'd think they'd put it in their docs somehow... 😄 

> Are the labels used to limit the fetch or does it fetch all issues and then filter? If all issues we can expect ~8 queries if it fetches 100 per page and then 22 for comments or closing?

I don't know, this is not explained properly in the documentation.... 😕 

It seems this is the final code: https://github.com/actions/stale/blob/main/dist/index.js
It has a class `IssuesProcessor` with a `processIssues` object.
See L271 in that file.

Maybe you can figure out what they do, from the code comments it appears that they do batching of some kind.

Reply:
> I don't think this would be a good idea, as all of their examples use the current cron job cycle of once every day at `01:30` hours. If they wanted normal users to run the action more often, I'd think they'd put it in their docs somehow... 😄

Not necessarily.

But anyway we can start like that and observe if we think it's "missing" too much. API rate limits are usually all calculated hourly

Reply:
> But anyway we can start like that and observe if we think it's "missing" too much. API rate limits are usually all calculated hourly

Sounds like a decent plan for now.